tensorflow_encrypted/config.py

```diff
@@ -56,6 +57,103 @@ def get_docker_cpu_quota() -> Optional[int]:
     return cpu_cores
 
 
+class TFESession():
+    """
+    Wrap a Tensorflow Session
+    """
+
+    def __init__(self, sess: tf.Session) -> None:
+        self.sess = sess
+
+    def __enter__(self):
+        self.sess = self.sess.__enter__()
+        return self
+
+    def __exit__(self, exec_type, exec_value, exec_tb):
+        self.sess.__exit__(exec_type, exec_value, exec_tb)
+
+    def sanitize_fetches(self, fetches: Any) -> List[Union[tf.Tensor, tf.Operation]]:
```

Comment:
What are `sanitize_fetches` and `decode_fetches` functions doing exactly? They're exposing hidden details of the underlying protocols and tensor types. For example, not all tensors currently have `.backing` members or `.from_decomposed` functions. Maybe they should but I think we should be leaving the decoding up to the protocol.

Reply:
Yes, indeed.
I wanted some feedback on how to generalize the behavior across all protocols

Reply:
I think I see what you're doing now. Not sure about the sanitized function but I think for decode we just need to expose a function in the protocol that knows how to decode. Sort of like the `.eval()` function on the `PondPublicTensor`.